htlcswitch/mock.go

```diff
@@ -32,16 +32,22 @@ import (
 type mockPreimageCache struct {
 	sync.Mutex
 	preimageMap map[[32]byte][]byte
+
+	// lookupDelay is used to delay the preimage lookup after the mutex is
+	// released.
+	lookupDelay time.Duration
```

Comment:
Can this delay be removed here and instead write an interceptor for the circuitmap, so that a channel can be exposed that is waited on before attempting to write the circuit?

Or a different approach to get rid of the delay?